src/cpp/database/database.hpp

```diff
@@ -514,6 +542,145 @@ class Database
     DatabaseDump dump_data_(
             const std::map<EntityId, ByteCountSample>& data);
 
+    /**
+     * @brief Check if 'entities_container' contains an 'id', throwing an exception if not.
+     *
+     * @param entities_container Reference to the dump of the entities to check.
+     * @param id id of the entity to check.
+     * @throws eprosima::statistics_backend::FileCorrupted if the 'id' does not exist in 'entities_container'.
+     */
+    void check_entity_container_contains_id(
+            DatabaseDump const& entities_container,
+            std::string const& id);
+
+    /**
+     * @brief Check that in the 'dump', the references of the entity iterator 'it' of type 'entity_tag'
+     * to entities of type 'reference_tag' are consistent and mutual. For this, the referenced entities must
+     * have reference to 'it' of type 'entity_tag'
+     *
+     * @param dump reference to the database dump.
+     * @param it iterator to the dump of the entity.
+     * @param entity_tag Type of the entity to check.
+     * @param reference_tag Type of the referenced entity to check.
+     * @throws eprosima::statistics_backend::FileCorrupted if the references are not consistent and mutual.
+     */
+    void check_entity_all_references(
+            DatabaseDump const& dump,
+            nlohmann::json::iterator const& it,
+            std::string const& entity_tag,
+            std::string const& reference_tag);
+
+    /**
+     * @brief Check that in the 'dump', the references of the entity iterator 'it' of type 'entity_tag'
+     * to entities of type 'reference_tag' is consistent and mutual. For this, the referenced entities must
+     * contain a reference to 'it' of type 'entity_tag'
+     *
+     * @param dump reference to the database dump.
+     * @param it iterator to the dump of the entity.
+     * @param entity_tag Type of the entity to check.
+     * @param reference_container_tag Type of the referenced entity container to check.
+     * @param reference_tag Type of the referenced entity to check.
+     * @throws eprosima::statistics_backend::FileCorrupted if the reference is not consistent and mutual.
+     */
```

Comment:
The only difference was that instead of "have reference" was "contains references", I agree that is not enough. I add an example to an easy visualization of the difference.